### PR TITLE
Add ability to share a single event

### PR DESCRIPTION
**Pull request by ay8s**
Adds share button to individual events if its in the future.

We should tidy up the text and url used.

### CODE REVIEW

**Comment by rodericj** [comment]
Maybe event specific urls? @jamescmartinez

File: SF iOS/SF iOS/EventDetails/EventDetailsViewController.m [hunk 1]
```diff
@@ -160,5 +167,20 @@ - (void)getTravelTimes {
...
 
+// MARK: Share
+
+- (void)share {
+    // TODO: Use the group name and coffeecoffeecoffee.coffee url.
+    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+    [formatter setDateStyle:NSDateFormatterMediumStyle];
+    [formatter setTimeStyle:NSDateFormatterShortStyle];
+    
+    NSString *text = [NSString stringWithFormat:@"Join us at %@ for coffee, kicking off %@.", self.event.venueName, [formatter stringFromDate:self.event.date]];
```
Comment (rodericj) [inline SF iOS/SF iOS/EventDetails/EventDetailsViewController.m]: Would it be beneficial to use the notification copy with the random seed through this API? https://github.com/ay8s/sf-ios/blob/task/event-share/SF%20iOS/SF%20iOS/Helpers/UserNotification/NSString%2BEventNotifications.h#L14

**Comment by rodericj** [comment]
I'm only testing on simulator at the moment but I'd assume that the text you have there shows up as the message when you're sharing to the twitters or something right?

**Comment by rodericj** [comment]
Did not mean to close that.

**Comment by ay8s** [comment]
@rodericj Yup. Most extensions will make use of the items as they see fit. In Twitters case it will use the text and append the URL at the end. Same for Buffer etc.

**Review by rodericj — CHANGES_REQUESTED** [review]
I would also say localization is probably pretty important here in addition to your TODOs

**Comment by ay8s** [comment]
Added group name and localization although we don't have a localizable strings file yet.

**Review by rodericj — APPROVED** [review]

File: SF iOS/SF iOS/EventDetails/EventDetailsViewController.m [hunk 2]
```diff
@@ -157,5 +166,21 @@ - (void)getTravelTimes {
     }];
 }
 
+// MARK: Share
+
+- (void)share {
+    // TODO: Use coffeecoffeecoffee.coffee url.
```
Comment (rodericj) [inline SF iOS/SF iOS/EventDetails/EventDetailsViewController.m:172]: @jamescmartinez do we have any concept of a c4 url for a specific event?

Comment (rodericj) [reply SF iOS/SF iOS/EventDetails/EventDetailsViewController.m:172]: @ay8s I just tried adding the URL and it takes away twitter as a share option. That's sort of lame. So I guess we either keep it as is or we get James to add the detail view on web

Comment (ay8s) [reply SF iOS/SF iOS/EventDetails/EventDetailsViewController.m:172]: We can just slot the URL into the text and fake it that way. 🤷‍♂ Damn Twitter!

Comment (rodericj) [reply SF iOS/SF iOS/EventDetails/EventDetailsViewController.m:172]: Yeah that didn't work for me when I tried it. Maybe I did it wrong.

**Pull request merged**